File: Projetos/Políticos/C++/sources/Senador.cpp
// Senador.cpp
#include <stdio.h>
#include <stdlib.h>
#include <iostream>
#include "../includes/Senador.h"
using namespace std;

Senador::Senador(string nome,string partido, string estado) {
    setNome(nome);
    setSalario(30000);
    setPartido(partido);
    setEstado(estado);
    setFuncao("propor no Senado leis federais em benefício da população.");
}
Senador::~Senador() {
    setNome("");
    setPartido("");
    setEstado("");
}
void Senador::apresentacao(){
    Politico::apresentacao();
    cout << "sou senador " << endl;
    cout << "Minha função é " << this->getFuncao() << endl;
    cout << "Fui eleito por " << this->getEstado() << endl;
    cout << "============================" << endl;
}

File: 07/Python3/Senador.py
#!/usr/bin/python3
# coding: utf-8
"""Classe Senador """

from Politico import Politico

class Senador(Politico):
    """ Classe Senador """

    def __init__(self, nome, partido, estado):
        """ Construtor da classe Senador"""
        Politico.__init__(self)
        self.set_nome(nome)
        self.set_partido(partido)
        self.set_estado(estado)
        self.set_funcao("propor no Senado leis em benefício da população.")

    def apresentacao(self):
        Politico.apresentacao(self)
        print ('sou senador ')
        print ('Minha função é ' + self.get_funcao())
        print ('Fui eleito por ' + self.get_estado())
        print ('============================')
